tests/callbacks/test_memory_monitor.py

```diff
@@ -10,9 +10,7 @@
 from tests.common import RandomClassificationDataset, SimpleModel, device
 
 
-@device('cpu', 'gpu')
 def test_memory_monitor_warnings_on_cpu_models(device: str):
-    # Error if the user sets device=cpu even when cuda is available
```

Comment:
Removed GPU test because it breaks things and doesn't really test anything meaningful.

All pytests share the same python process, and we've already initialized a NCCL backend. If we try running a unit test with device=cpu in the GPU test suite, we get an error trying to initialize a GLOO backend on a python process which already has a NCCL process initialized. This is only an issue because all tests run in the same process.